wandb/sdk/wandb_run.py

```diff
@@ -2964,52 +2959,7 @@ def link_artifact(
             # TODO: implement offline mode + sync
             raise NotImplementedError
 
-        # Wait until the artifact is committed before trying to link it.
-        artifact.wait()
-
-        organization = ""
-        if is_artifact_registry_project(project):
-            organization = entity or self.settings.organization or ""
-            # In a Registry linking, the entity is used to fetch the organization of the artifact
-            # therefore the source artifact's entity is passed to the backend
-            entity = artifact._source_entity
-        project = project or self.project
-        entity = entity or self.entity
-        handle = self._backend.interface.deliver_link_artifact(
-            artifact,
-            portfolio,
-            aliases,
-            entity,
-            project,
-            organization,
-        )
-        if artifact._ttl_duration_seconds is not None:
-            wandb.termwarn(
-                "Artifact TTL will be disabled for source artifacts that are linked to portfolios."
-            )
-        result = handle.wait_or(timeout=None)
-        response = result.response.link_artifact_response
-        if response.error_message:
-            wandb.termerror(response.error_message)
-            return None
-        if response.version_index is None:
-            wandb.termerror(
-                "Error fetching the linked artifact's version index after linking"
-            )
-            return None
-
-        try:
-            artifact_name = f"{entity}/{project}/{portfolio}:v{response.version_index}"
-            if is_artifact_registry_project(project):
-                if organization:
-                    artifact_name = f"{organization}/{project}/{portfolio}:v{response.version_index}"
-                else:
-                    artifact_name = f"{project}/{portfolio}:v{response.version_index}"
-            linked_artifact = self._public_api()._artifact(artifact_name)
-        except Exception as e:
-            wandb.termerror(f"Error fetching link artifact after linking: {e}")
-            return None
-        return linked_artifact
+        return artifact.link(target_path, aliases)
```

Comment:
oh this is interesting! so we're not really duplicating code then. this removes the dependency of the link action on wandb-core/internal_api altogether, right?